### PR TITLE
Create <List.LinkItem />

### PR DESCRIPTION
**Pull request by tcK1**
We want to be able to click on the full area of the item. To do that, we need to apply the styles directly to the anchor tag.

Since we need this, we should create a new compound.

### CODE REVIEW

File: packages/doc/content/components/components/list/list-web.mdx [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 Gympass `<List />` component implements all our design guidelines, be sure to wrap your items with `<List.Item/>` that uses default spacing, border and color pallete tokens.
 
+If you need to use a anchor there's also the `<List.LinkItem/>` available.
```
Comment (ggdaltoso) [inline packages/doc/content/components/components/list/list-web.mdx]: If you need to use an anchor, `<List.LinkItem/>` is also available

💅

File: packages/yoga/src/List/index.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import { List, ListItem } from './web';
+import { List, ListItem, ListLinkItem } from './web';
 
 List.Item = ListItem;
+List.LinkItem = ListLinkItem;
```
Comment (ggdaltoso) [inline packages/yoga/src/List/index.js]:
```suggestion
import { List, ListItem, LinkItem } from './web';

List.Item = ListItem;
List.LinkItem = LinkItem;
```

File: packages/yoga/src/List/web/List.test.jsx [hunk 3]
```diff
@@ -98,6 +98,19 @@ describe('<List />', () => {
         );
         expect(container).toMatchSnapshot();
       });
+
+      it('should match snapshot with a default list and using an link item', () => {
```
Comment (ggdaltoso) [inline packages/yoga/src/List/web/List.test.jsx]:
```suggestion
      it('should match snapshot with a default list and using a link item', () => {
```

File: packages/yoga/src/List/web/ListLinkItem.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
```
Comment (ggdaltoso) [inline packages/yoga/src/List/web/ListLinkItem.jsx]: Could you rename this file to reflect its component name?

`LinkItem.jsx`

File: packages/yoga/src/List/web/ListLinkItem.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import styled from 'styled-components';
+
+const StyledItem = styled.a`
```
Comment (ggdaltoso) [inline packages/yoga/src/List/web/ListLinkItem.jsx]:
```suggestion
const StyledLinkItem = styled.a`
```

File: packages/yoga/src/List/web/ListLinkItem.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+const ListItem = ({ ...rest }) => (
+  <li>
+    <StyledItem {...rest} />
+  </li>
+);
+
+ListItem.displayName = 'List.Item';
+
+export default ListItem;
```
Comment (ggdaltoso) [inline packages/yoga/src/List/web/ListLinkItem.jsx]:
```suggestion
const LinkItem = ({ ...rest }) => (
  <li>
    <StyledLinkItem {...rest} />
  </li>
);

LinkItem.displayName = 'List.LinkItem';

export default LinkItem;
```

File: packages/yoga/src/List/web/index.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
+import ListLinkItem from './ListLinkItem';
```
Comment (ggdaltoso) [inline packages/yoga/src/List/web/index.js]:
```suggestion
import ListLinkItem from './LinkItem';
```

File: packages/yoga/src/List/web/index.js [hunk 8]
```diff
@@ -1,4 +1,5 @@
...
 import ListItem from './ListItem';
 import List from './List';
 
-export { List, ListItem };
+export { List, ListItem, ListLinkItem };
```
Comment (ggdaltoso) [inline packages/yoga/src/List/web/index.js]:
```suggestion
export { List, ListItem, LinkItem };
```

File: packages/yoga/src/List/web/LinkItem.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      ${list.listItem.padding.bottom}px
+      ${list.listItem.padding.left}px;
+  `}
+
+  display: inline-block;
+  width: 100%;
+  height: 100%;
+
+  text-decoration: none;
+`;
```
Comment (victormath12) [inline packages/yoga/src/List/web/LinkItem.jsx]: In all styled components we use a CSS Ordering (we didn't explained well in the DOJO, sorry).
What do you think to learn and use in this styled ?

**This is the reference:**
https://webdesign.tutsplus.com/articles/outside-in-ordering-css-properties-by-importance--cms-21685

File: packages/doc/content/components/components/list/list-web.mdx [hunk 10]
```diff
@@ -43,6 +45,10 @@ render(() => {
           />
         </SwitchLine>
       </List.Item>
+
+      <List.LinkItem href="#">
```
Comment (victormath12) [inline packages/doc/content/components/components/list/list-web.mdx]: What do you think to use a real link in the example?

```suggestion
      <List.LinkItem href="https://www.gympass.com/" target="_blank">
```

File: packages/yoga/src/List/web/LinkItem.jsx (hunk 9 above)
Comment (victormath12) [reply packages/yoga/src/List/web/LinkItem.jsx]: And we use this callback 

```
${({
    theme: {
      yoga: {
        components: { list },
      },
    },
  }) => `
    padding:
      ${list.listItem.padding.top}px
      ${list.listItem.padding.right}px
      ${list.listItem.padding.bottom}px
      ${list.listItem.padding.left}px;
  `}
```

in the end of all rules of styled component

File: packages/yoga/src/List/web/LinkItem.jsx [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+      ${list.listItem.padding.bottom}px
+      ${list.listItem.padding.left}px;
+  `}
+
+  display: inline-block;
+
+  width: 100%;
+  height: 100%;
+
+  text-decoration: none;
```
Comment (victormath12) [inline packages/yoga/src/List/web/LinkItem.jsx]: We use this callback in the end of all of styled component's rules.

```suggestion
  display: inline-block;

  width: 100%;
  height: 100%;

  text-decoration: none;
  
  ${({
    theme: {
      yoga: {
        components: { list },
      },
    },
  }) => `
    padding:
      ${list.listItem.padding.top}px
      ${list.listItem.padding.right}px
      ${list.listItem.padding.bottom}px
      ${list.listItem.padding.left}px;
  `}
```

**Pull request merged**